### PR TITLE
[metrics] count empty range in gc

### PR DESCRIPTION
**Pull request by AndreMouche**
@zhangjinpeng1987 @disksing PTAL

### CODE REVIEW

File: src/storage/metrics.rs [hunk 1]
```diff
@@ -76,4 +76,10 @@ lazy_static! {
             "Total number of rawkv commands received.",
             &["type"]
         ).unwrap();
+
+    pub static ref GC_EMPTY_RANGE_COUNTER: Counter =
+        register_counter!(
+            "gc_empty_range_total",
```
Comment (siddontang) [inline src/storage/metrics.rs]: tikv_storage_gc_empty_range_total

File: src/storage/txn/scheduler.rs [hunk 2]
```diff
@@ -369,12 +369,17 @@ fn process_read(cid: u64, mut cmd: Command, ch: SendCh<Msg>, snapshot: Box<Snaps
         Command::Gc { ref ctx, safe_point, ref mut scan_key, .. } => {
             let mut reader =
                 MvccReader::new(snapshot.as_ref(), Some(ScanMode::Forward), true, None);
+            let start_key = scan_key.clone();
```
Comment (siddontang) [inline src/storage/txn/scheduler.rs]: scan_key is always the start_key here?

**Comment by siddontang** [comment]
PTAL @hhkbp2

File: src/storage/txn/scheduler.rs (hunk 2 above)
Comment (AndreMouche) [inline src/storage/txn/scheduler.rs]: YES. Range start gc with scan_key=none, and scan_key will be used as start key(not none) in later steps.

Comment (siddontang) [inline src/storage/txn/scheduler.rs]: got it, please add a comment here.

File: src/storage/txn/scheduler.rs [hunk 3]
```diff
@@ -369,12 +369,18 @@ fn process_read(cid: u64, mut cmd: Command, ch: SendCh<Msg>, snapshot: Box<Snaps
         Command::Gc { ref ctx, safe_point, ref mut scan_key, .. } => {
             let mut reader =
                 MvccReader::new(snapshot.as_ref(), Some(ScanMode::Forward), true, None);
+            // scan_key is used as start_key here,and Range start gc with scan_key=none.
+            let start_key = scan_key.clone();
```
Comment (disksing) [inline src/storage/txn/scheduler.rs]: let is_start_key = scan_key.is_none()
you don't have to clone it.

Comment (AndreMouche) [inline src/storage/txn/scheduler.rs]: ok

**Comment by disksing** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**